apps/server/src/modules/task/service/task.service.ts

```diff
@@ -35,7 +35,13 @@ export class TaskService {
 		filters?: { draft?: boolean; noFutureAvailableDate?: boolean },
 		options?: IFindOptions<Task>
 	): Promise<Counted<Task[]>> {
-		return this.taskRepo.findBySingleParent(creatorId, courseId, filters, options);
+		const repoFilters: { draft?: boolean; noFutureAvailableDate?: boolean; userId?: EntityId } = { ...filters };
+		const user = await this.authorizationService.getUserWithPermissions(creatorId);
+		if (!this.authorizationService.hasAllPermissions(user, [Permission.TASK_DASHBOARD_TEACHER_VIEW_V3])) {
```

Comment:
Authorisation must be part of the UC.

Reply:
Is it is possible to add 2 service methodes and the UC make the split which is called, depend on the permission that the user has? I want to avoid that the authorisation service is needed in service layer.

Reply:
we have some tech depth in the task service, where the authorization service is already needed for other stuff too.
I would rather refactor it together but separate from this ticket

Reply:
Ok if you have this it is ok. In the end the authorisation service should not be use in service layer. But it must not be done in this ticket.